src/libcore/iter.rs

```diff
@@ -369,6 +369,25 @@ pub trait Iterator {
     /// `None` here means that either there is no known upper bound, or the
     /// upper bound is larger than `usize`.
     ///
+    /// # Safety
+    ///
+    /// Implementations of this trait are not guaranteed to yield the
+    /// declared number of elements. The iterator may yield less than
+    /// lower bound or more than upper bound of elements.
+    ///
+    /// `size_hint()` can be used to optimize the fast path, and can
+    /// be used to panic if estimation is incorrect, but must not be
+    /// used, e. g. to omit bound checks in unsafe operations. Incorrect
+    /// estimation of size must not break memory safety.
+    ///
+    /// That said, the implementation should provide proper estimation,
+    /// because otherwize it may hurt performance or lead to a panic.
```

Comment:
spelling: *otherwise*

I'd like to use the phrase *provide a correct estimation*. It may hurt performance or break the trait's protocol.